hooks/useLogin.ts

```diff
@@ -18,27 +24,30 @@ export enum STATUS {
 // {issuer, publicAddress, email}
 // https://github.com/magiclabs/magic-js/blob/master/packages/%40magic-sdk/types/src/modules/user-types.ts#L17-L21
 
-// const $metadata = useLogin('/go-somewhere-if-it-does-not-work')
+// const $metadata = useLogin({redirect: '/go-somewhere-if-it-does-not-work'})
 
 export interface LoginProps {
   redirect?: string
-  timeout?: number
 }
+
 export function useLogin(config: LoginProps = {}) {
-  const { redirect, timeout = -1 } = config
+  const $router = useRouter()
```

Comment:
This use of `useRouter` is triggering additional /me data fetches and executions of the `checkLoggedIn` useEffect when the router is added to the useEffect's dependencies, so we should switch back to Router here.

Next.js recommends using the global Router in useEffects and event handlers, and the useRouter/withRouter in components: https://github.com/vercel/next.js/discussions/18522#discussioncomment-836362

Another related issue https://github.com/vercel/next.js/issues/18127

Reply:
I tried changing to the static `Router.push` but it reverted all of core intent of this PR. (That is, when swapped, the page doesn't update on hitting `/leaderboard` after being redirected from `/login`.) So I am intending to keep this change.